### PR TITLE
chore: prep for v0.14.0

### PR DESCRIPTION
**Pull request by sipsma**

### CODE REVIEW

File: .changes/v0.14.0.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## v0.14.0 - 2024-11-08
+
+
+### 🔥 Breaking Changes
+- Rename `dagger-engine` core GraphQL query to `engine` by @Marcos Lilljedahl in https://github.com/dagger/dagger/pull/8568
```
Comment (sipsma) [inline .changes/v0.14.0.md:5]: Oops, just realized this is gonna be a random Marcos 😄, will fix to be correct gh username

File: sdk/python/.changes/v0.14.0.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  For example:
+  ```toml
+  # pyproject.toml
+  
+  [project.entry-points."dagger.mod"]
+  main_object = "my_module.main:MyModule"
+  ```
+
+### Deprecated
+- In modules, the `index-url` and `index-extra-url` settings in `[tool.uv]` are now deprecated in favor of the new `[[tool.uv.index]]` setting by @helderco in https://github.com/dagger/dagger/pull/8772 \
```
Comment (sipsma) [inline sdk/python/.changes/v0.14.0.md:22]: typo with `\` at the end, will fix

**Comment by sipsma** [comment]
Huh there's some legit failures in the go sdk linting, looks related to enum changes? Seeing if it just needs a regen..

**Review by vito — APPROVED** [review]

**Comment by sipsma** [comment]
> Huh there's some legit failures in the go sdk linting, looks related to enum changes? Seeing if it just needs a regen..

Okay yeah it's https://github.com/dagger/dagger/pull/8669, which kicks in on the Go SDK now that we hit v0.14.0.

This is okay, but it's an extra breaking change of something that has only been deprecated for a single patch release and also doesn't have any changelog associated with it since it just magically happens when the version number hits v0.14.0...

I'll double check with the team on what approach we want to go with here. At min need to leave in but add changelogs. Can consider leaving deprecated until v0.15.0 too since it's barely been deprecated for a week now.

**Pull request merged**